### PR TITLE
Added support for Solus

### PR DESCRIPTION
**Pull request by AktasC**
I tried different things but this is the best out of it.
I would have loved to integer fish instead of zsh, in my opinion it's more complete ..
Also, I don't think /usr/bin/zsh works since, for me at least, it was always found at /bin/zsh ..
One more thing to note is that we need "sudo" to use "chsh" in Solus.

If you need more details or need me to modify some stuff just ask for it and I'll try to do it right away :)

### CODE REVIEW

File: installation/install [hunk 1]
```diff
@@ -230,20 +233,35 @@ script_header "INSTALLATION DE EMACS ET DU SYSTÈME DE HEADERS EPITECH"
...
+	sys_install build-essential
+fi
+
 script_header "INSTALLATION DE GCC"
 sys_install gcc
 
-script_header "INSTALLATION DE BUILD-ESSENTIAL POUR LA COMPILATION"
-sys_install build-essential
+script_header "INSTALLATION DE MAKE"
+sys_install make
```
Comment (bew) [inline installation/install:246]: as `make` is also used for compilation, you should put his installation in the block named `INSTALLATION DE BUILD-ESSENTIAL POUR COMPILER`

In fact, I would write it like (before the GCC block):
```sh
script_header	"INSTALLATION DES OUTILS POUR LA COMPILATION"
if "$os" != "solus"; then
    sys_install build-essential
fi
sys_install make
```
Feel free to make it like this, or to justify your choices

File: installation/install [hunk 2]
```diff
@@ -230,20 +233,35 @@ script_header "INSTALLATION DE EMACS ET DU SYSTÈME DE HEADERS EPITECH"
...
+	sys_install libncurses5
+fi
+
+if "$os" == "solus"; then
+	script_header	"INSTALLATION DE LA LIB NCURSES"
+	sys_install ncurses-devel
+fi
+
+script_header "INSTALLATION DE G++"
+sys_install g++
```
Comment (bew) [inline installation/install:265]: You should put the installation of `g++` next the compilation of `gcc` (you can of course rename the block to mention the 2)

File: installation/install [hunk 3]
```diff
@@ -230,20 +233,35 @@ script_header "INSTALLATION DE EMACS ET DU SYSTÈME DE HEADERS EPITECH"
...
-sys_install libncurses5
+if "$os" != "solus"; then
+	script_header	"INSTALLATION DE LA LIB NCURSES"
+	sys_install libncurses5
+fi
+
+if "$os" == "solus"; then
+	script_header	"INSTALLATION DE LA LIB NCURSES"
+	sys_install ncurses-devel
+fi
```
Comment (bew) [inline installation/install:262]: the condition against `solus` is only on the package installation, you should have only 1 instruction for the `script_header` and do the condition around the `sys_install` command instead, with a branching like:
```
if ...; then
    # ...
else
    # ...
fi
```

**Comment by bew** [comment]
> I would have loved to integer fish instead of zsh, in my opinion it's more complete ..

I don't agree, but you have the right to be wrong :)

I'd say that out-of-the-box fish is prettier, but I don't think it's more powerful.. (can you make a snake game, or a FTP server using only fish (without external programs like curl..) ? In zsh you can !)

> Also, I don't think /usr/bin/zsh works since, for me at least, it was always found at /bin/zsh ..

The final binaries can be in different places, but I don't think it's a reason to disable zsh installation (you could make a condition around `chsh` for solus)

What do you think ?

File: installation/install [hunk 4]
```diff
@@ -85,11 +81,12 @@ function handle_error
 
 function get_os_type
 {
-	which zypper &> /dev/null && os="opensuse"
 	which pacman &> /dev/null && os="archlinux"
-	which dnf &> /dev/null && os="fedora"
 	which apt-get &> /dev/null && os="debian"
+	which dnf &> /dev/null && os="fedora"
 	which emerge &> /dev/null && os="gentoo"
+	which zypper &> /dev/null && os="opensuse"
```
Comment (bew) [inline installation/install:88]: Why did you move opensuse & fedora position ?

Basically, when you make a pull request, you should modify or add lines related with the issue you are resolving

In this case, you should not move/change lines that are not related to `add support for solus`.

File: installation/install [hunk 5]
```diff
@@ -115,21 +112,24 @@ function sys_upgrade
...
 			;;
+		fedora)
+			sudo dnf update
+			;;
 		gentoo)
 			sudo emerge -u world
 			;;
+		opensuse)
+			sudo zypper update
+			;;
```
Comment (bew) [inline installation/install:129]: Same comment as above on the movement of lines, not related to you issue

If you want to move or refactor codes, you should open an other issue for that, and we will discuss it

File: installation/install [hunk 6]
```diff
@@ -141,21 +141,24 @@ function sys_install
...
 				;;
+			fedora)
+				echo "dnf install"
+				;;
 			gentoo)
 				echo "emerge"
 				;;
+			opensuse)
+				echo "zypper install"
+				;;
```
Comment (bew) [inline installation/install:158]: Same comment on moving lines

File: installation/install [hunk 7]
```diff
@@ -264,16 +283,24 @@ handle_error $?
 
 script_header "INSTALLATION ET CONFIGURATION DE ZSH ET OH MY ZSH"
 
-script_header "ZSH" $C_YELLOW
-sys_install zsh
+if "$os" != "solus"; then
+	script_header "ZSH" $C_YELLOW
+	sys_install zsh
+fi
```
Comment (bew) [inline installation/install:289]: So you disable ZSH installation for `solus` ?

File: installation/install [hunk 8]
```diff
@@ -264,16 +283,24 @@ handle_error $?
...
+	sudo chsh $USER -s /bin/zsh
+	handle_error $?
+	echo "alias blih='blih -u $1'" >> /home/$USER/.zshrc
+	echo "alias ns_auth='ns_auth -u $1'" >> /home/$USER/.zshrc
+	echo "alias ne='emacs -nw'" >> /home/$USER/.zshrc
+fi
+
+echo "alias blih='blih -u $1'" >> /home/$USER/.bashrc
+echo "alias ns_auth='ns_auth -u $1'" >> /home/$USER/.bashrc
+echo "alias ne='emacs -nw'" >> /home/$USER/.bashrc
```
Comment (bew) [inline installation/install:303]: to reduce code duplication, maybe you could populate a `~/.aliases.sh` file with aliases, and source it in `~/.bashrc` & `~/.zshrc`

**Review by bew — CHANGES_REQUESTED** [review]

File: installation/install (hunk 4 above)
Comment (AktasC) [reply installation/install:88]: I just changed the positions so that it gets in the alphabetical order.

**Comment by AktasC** [comment]
Thanks for your review.
I will modify and then push again my job when it's totally done.

As for ZSH : 
I've used it for one year at epitech then I swaped for fish which I found better for auto-completion and other stuff without having to configure anything.
In my opinion, a tool that you have to configure in order to get what you could with another isn't a good tool.
Plug'n'Play is kinda better for me since I'm jumping from one OS to the other every week.
I used to have my Arch Distro on Reddit's UnixPorn section and it pleased a lot of people.
For now, I will focus on improving your script and will certainly also add support for Deepin (since a lot of people use it in my school currently).

**Comment by AktasC** [comment]
Thanks for your review.
I will modify and then push again my job when it's totally done.

As for ZSH : 
I've used it for one year at epitech then I swaped for fish which I found better for auto-completion and other stuff without having to configure anything.
In my opinion, a tool that you have to configure in order to get what you could with another isn't a good tool.
Plug'n'Play is kinda better for me since I'm jumping from one OS to the other every week.
I used to have my Arch Distro on Reddit's UnixPorn section and it pleased a lot of people.
For now, I will focus on improving your script and will certainly also add support for Deepin (since a lot of people use it in my school currently).

**Comment by bew** [comment]
Why did you closed this PR ? You can  continue tu push on your branch and the PR here will update accordingly

File: installation/install (hunk 4 above)
Comment (bew) [reply installation/install:88]: Oh nice, I understand & like that, let's keep it that way then! 

I think you should have made a commit which mentioned this, so there would be no misunderstanding

**Comment by bew** [comment]
> In my opinion, a tool that you have to configure in order to get what you could with another isn't a good tool.

I don't agree but I understand your point. 

Deepin..  don't know, but it's great, thank you!

**Comment by AktasC** [comment]
I closed it because I made another pull request, didn't wanna pollute here..
We both know what Deepin is worth but .. Yeah .. Some people like it.

**Pull request closed without merging**